src/index.js

```diff
@@ -1,22 +1,40 @@
-import React, { useState } from "react";
-import AppNavigator from "./navigation/navigators/AppNavigator";
+import React, { useState, useEffect } from "react";
+import AsyncStorage from "@react-native-community/async-storage";
+import {
+  AppExtendedContainer,
+  AppPrimaryContainer
+} from "./navigation/navigators/AppNavigator";
 import TranslationProvider, {
   Translation
 } from "./components/TranslationProvider";
 import { changeLocale } from "./i18n";
 
-const AppContainer = () => {
+const AppLocalised = () => {
   const [locale, changeLocaleState] = useState("ru");
+  const [isFirstLaunch, toggleFirstLaunch] = useState(true);
+  let UltimateNavigator = AppExtendedContainer;
+
   const onLocaleChange = localeKey => {
     changeLocale(localeKey);
     changeLocaleState(localeKey);
   };
 
+  useEffect(() => {
+    AsyncStorage.getItem("lang").then(langSet => {
+      toggleFirstLaunch(langSet);
+      onLocaleChange(langSet);
+    });
+  });
+
+  if (isFirstLaunch) {
+    UltimateNavigator = AppPrimaryContainer;
```

Comment:
For me it would be preferable to set name of navigator with first launch screens kind of FirstLaunchNavigator, so that it's clear what this navigator is served for

Reply:
fixed it!